### ISSUE TITLE
set only property bug

### ISSUE THREAD

**Issue by RichardEll** [Microsoft/TypeScript#4839]
If you define a class that has a property which only has a setter, but not a getter, and then attempt to use the non-existent getter the code will compile, but the getter is undefined, leading to a crash when it is used. For example:
```
class MyClass {

    private theCount: number;

    constructor() {
        this.theCount = 5;
    }

    public set count(i: number) {
        this.theCount = i;
    }
}

window.onload = () => {
    var obj: MyClass = new MyClass();
    obj.count = 12;
    var num: number = obj.count;
    document.getElementById("content").innerHTML = num.toString();
};
```
The above will compile, but crashes when reading obj.count.
Surely it would be better if this code did not compile.

**Comment by mhegazy** [comment]
se #12 for more information.

**Comment by RichardEll** [comment]
Is this really a duplicate? #12 is a suggestion for a read only modifier. I am not advocating that. I just think that if no getter is defined then the compiler should flag an error at compile time if you attempt to use it (like C# does).

**Comment by kitsonk** [comment]
Write only property descriptors are perfectly valid JavaScript, therefore should be perfectly valid TypeScript.  While it may not be a very useful programming paradigm, it is a legitimate one.

**Comment by kitsonk** [comment]
Actually, I see what you want, you want #295 which is to throw when you attempt to access a write-only property (or a read-only property).  As @RyanCavanaugh states, the type system doesn't currently support that differentiation.

**Comment by RichardEll** [comment]
I see the problem.
I discovered this little quirk when I made a class that allowed a particular property (something like a shape dimension - I forget) to be changed. At the time I didn't see the need for the ability to read the value back and, being lazy, didn't bother to implement it. But, some time later, I wrote some code which did attempt to read the value back and was rather puzzled when the code crashed out when it read undefined. I'm pretty sure this won't be the last time I do this sort of thing and it would be great if the compiler would stop me doing this. But I appreciate there are technical reasons why this is difficult.
